app/content/profile/index.js

```diff
@@ -1,29 +1,56 @@
 import { useNavigation } from "expo-router";
+import {
+  collection,
+  doc,
+  getDoc,
+  getDocs,
+  query,
+  where,
+} from "firebase/firestore";
+import { useContext, useEffect, useState } from "react";
 import { ScrollView, StyleSheet, Text, View } from "react-native";
 import { Appbar, PaperProvider } from "react-native-paper";
 import { SafeAreaView } from "react-native-safe-area-context";
+import { CurrentUserContext } from "~/Contexts";
 import DrillCard from "~/components/drillCard";
 import ProfileCard from "~/components/profileCard";
-import drillData from "~/drill_data.json";
+import db from "~/firebaseConfig";
 
 function Index(props) {
   const navigation = useNavigation();
-  const user = drillData["teams"]["1"]["users"]["1"];
-  const drills = drillData["teams"]["1"]["drills"];
-  const attemptedDrills = user["history"]
-    ? Object.keys(drills).filter((drillId) => drillId in user["history"])
-    : [];
+  const [userData, setUserData] = useState({});
+  const userId = useContext(CurrentUserContext)["currentUser"];
+  const [drills, setDrills] = useState({});
+
+  useEffect(() => {
+    let uniqueDrills = new Set();
+    getDoc(doc(db, "teams", "1", "users", userId)).then((doc) => {
+      setUserData(doc.data());
+    });
+    getDocs(
+      query(collection(db, "teams", "1", "attempts")),
```

Comment:
If the only info collected from the `team > attempts` object is the unique drill ID's, maybe it would be more performant / less requests if we stored a list of drill ID's the player has attempted in `team > users > uid` object, and then update if needed from drill submission.

Just an idea though

Reply:
I think the querying is both so fast and so slow that it wouldn't make a tangible difference. It's more impactful to get the caching behavior right.
Although I'm down to do a test.